plugins/experimental/remap_purge/remap_purge.c

```diff
@@ -202,34 +202,38 @@ handle_purge(TSHttpTxn txnp, PurgeInstance *purge)
   bool should_purge = false;
 
   if (TS_SUCCESS == TSHttpTxnClientReqGet(txnp, &reqp, &hdr_loc)) {
-    /* First see if we require the "secret" to be passed in a header, and then use that */
-    if (purge->header) {
-      TSMLoc field_loc = TSMimeHdrFieldFind(reqp, hdr_loc, purge->header, purge->header_len);
-
-      if (field_loc) {
-        const char *header;
-        int header_len;
-
-        header = TSMimeHdrFieldValueStringGet(reqp, hdr_loc, field_loc, -1, &header_len);
-        TSDebug(PLUGIN_NAME, "Checking for %.*s == %s ?", header_len, header, purge->secret);
-        if (header && (header_len == purge->secret_len) && !memcmp(header, purge->secret, header_len)) {
-          should_purge = true;
+    int method_len     = 0;
+    const char *method = TSHttpHdrMethodGet(reqp, hdr_loc, &method_len);
+
+    if ((TS_HTTP_METHOD_PURGE == method) || ((TS_HTTP_METHOD_GET == method) && purge->allow_get)) {
+      /* First see if we require the "secret" to be passed in a header, and then use that */
+      if (purge->header) {
+        TSMLoc field_loc = TSMimeHdrFieldFind(reqp, hdr_loc, purge->header, purge->header_len);
+
+        if (field_loc) {
+          const char *header;
+          int header_len;
+
+          header = TSMimeHdrFieldValueStringGet(reqp, hdr_loc, field_loc, -1, &header_len);
+          TSDebug(PLUGIN_NAME, "Checking for %.*s == %s ?", header_len, header, purge->secret);
+          if (header && (header_len == purge->secret_len) && !memcmp(header, purge->secret, header_len)) {
+            should_purge = true;
+          }
+          TSHandleMLocRelease(reqp, hdr_loc, field_loc);
         }
-        TSHandleMLocRelease(reqp, hdr_loc, field_loc);
-      }
-    } else {
-      /* We are matching on the path component instead of a header */
-      if (TS_SUCCESS == TSHttpHdrUrlGet(reqp, hdr_loc, &url_loc)) {
-        int path_len = 0, method_len = 0;
-        const char *path   = TSUrlPathGet(reqp, url_loc, &path_len);
-        const char *method = TSHttpHdrMethodGet(reqp, hdr_loc, &method_len);
-
-        TSDebug(PLUGIN_NAME, "Checking PATH = %.*s", path_len, path);
-        if (((TS_HTTP_METHOD_PURGE == method) || ((TS_HTTP_METHOD_GET == method) && purge->allow_get)) && path &&
-            (path_len >= purge->secret_len) && !memcmp(path + (path_len - purge->secret_len), purge->secret, purge->secret_len)) {
-          should_purge = true;
+      } else {
+        /* We are matching on the path component instead of a header */
+        if (TS_SUCCESS == TSHttpHdrUrlGet(reqp, hdr_loc, &url_loc)) {
+          int path_len     = 0;
+          const char *path = TSUrlPathGet(reqp, url_loc, &path_len);
+
+          TSDebug(PLUGIN_NAME, "Checking PATH = %.*s", path_len, path);
+          if (path && (path_len >= purge->secret_len) &&
+              !memcmp(path + (path_len - purge->secret_len), purge->secret, purge->secret_len)) {
```

Comment:
Suggest comparing the secret to the result of ``strrchr(path, '/')`` or similar. This would succeed if the secret is "bar" and the URL is "/path/to/foobar".